mozjs/src/rust.rs

```diff
@@ -433,6 +443,35 @@ impl Drop for Runtime {
     }
 }
 
+/// JSContext that is Send and Sync
+/// but supports only limited (thread safe) operations
+pub struct SyncJSContext(Arc<Mutex<Option<*mut JSContext>>>);
```

Comment:
I like the name using "ThreadSafe" a bit better here as it matches the terminology we use for DOM nodes in layout. `Sync` maybe leans to heavily on the sense of "locking."
 
```suggestion
/// A version of the [`JSContext`] that can be used from other threads and is thus
/// `Send` and `Sync`. This should only ever expose operations that are marked as
/// thread-safe by the SpiderMonkey API.
/// See <https://searchfox.org/mozilla-central/rev/7a85a111b5f42cdc07f438e36f9597c4c6dc1d48/js/src/vm/JSContext.h#848>.
pub(crate) struct ThreadSafeJSContext(Arc<Mutex<Option<*mut JSContext>>>);
```

Reply:
We need to be able to use this struct in servo so it needs to be pub.

Reply:
Oh sorry! That makes sense. I thought this change was in Servo itself. Too many repositories.